feat: add `pre-handlers` package

mohammadhonarvar opened the pull request.

mohammadhonarvar committed:
feat: add `pre-handlers` package

mohammadhonarvar committed:
feat(nanotron): use `pre-handlers` package

mohammadhonarvar committed:
deps: update

mohammadhonarvar committed:
deps: update

alimd commented on packages/pre-handlers/package.json:
```diff
@@ -0,0 +1,75 @@
+{
+  "name": "@alwatr/pre-handlers",
+  "version": "1.0.0",
```
-0

alimd commented on packages/pre-handlers/src/main.ts:
```diff
@@ -0,0 +1,5 @@
+
+export * from './parse-body-as-json.js';
+
+__dev_mode__: packageTracer.add(__package_name__, __package_version__);
```
این جاش تو همون لاگر هستش

alimd commented on packages/pre-handlers/src/parse-body-as-json.ts:
```diff
@@ -0,0 +1,36 @@
+    });
+  }
+
+  logger.logMethodArgs?.('parseBodyAsJson', {body: this.sharedMeta.body});
```
بادی که آرگ نیست!
این چه مدلیه
اگر میخوای باید logProperty کنی یا methodFull

alimd commented on packages/pre-handlers/package.json:
```diff
@@ -0,0 +1,75 @@
+  },
+  "dependencies": {
+    "@alwatr/nanolib": "^1.3.0",
+    "@alwatr/nanotron-api-server": "workspace:^"
```
نیازش به api-server در حد تایپ هستش پس فقط مال بیلده و میشه devDeps

alimd commented on packages/pre-handlers/src/parse-body-as-json.ts:
```diff
@@ -0,0 +1,36 @@
+export async function parseBodyAsJson(
+  this: NanotronClientRequest<{body: DictionaryOpt}>,
+): Promise<void> {
+  const bodyBuffer = await this.getBodyRaw();
```
log

alimd reviewed (changes requested):
plz

mohammadhonarvar committed:
fix(pre-handlers): some issues of `review` feedbacks

Co-authored-by: Ali Mihandoost <ali.mihandoost@gmail.com>

mohammadhonarvar committed:
deps: update

Co-authored-by: Ali Mihandoost <ali.mihandoost@gmail.com>

mohammadhonarvar committed:
deps: update

alimd committed:
docs(pre-handlers): add docs with example

alimd committed:
chore: Update extension in .vscode/extensions.json

alimd commented on packages/pre-handlers/src/parse-body-as-json.ts:
```diff
@@ -0,0 +1,38 @@
+  this: NanotronClientRequest<{body: DictionaryOpt}>,
+): Promise<void> {
+  const bodyBuffer = await this.getBodyRaw();
+  logger.logProperty?.('bodyBuffer', bodyBuffer);
```
😢

alimd reviewed (changes requested):
!

alimd committed:
refactor(pre-handlers): Rename parse-body-as-json.ts to handler/parse-body-as-json.ts

alimd committed:
feat(pre-handlers): Add requireAccessToken middleware

This commit adds a new file, require-access-token.ts, which contains a middleware function for Nanotron API requests. The function checks the authorization header for a Bearer token and compares it to the provided access token. If the token is missing or invalid, it sends an appropriate error response and prevents further handlers from executing.

Example usage:
nanotronApiServer.defineRoute({
  method: 'POST',
  url: 'secure-endpoint',
  preHandlers: [requireAccessToken('mySecretToken')],
  async handler() {
    this.serverResponse.replyJson({
      ok: true,
      message: 'Access granted!',
    });
  },
});

alimd committed:
feat(pre-handlers): Add getAuthBearer function

This commit adds the getAuthBearer function to the pre-handlers package. The function extracts the Bearer token from the authorization header and returns it. If the header is invalid or the token is missing, it returns null.

Example usage:
console.log(this.headers.authorization); // 'Bearer myToken123'
const token = getAuthBearer(this.headers.authorization);
console.log(token); // 'myToken123'

const invalidToken = getAuthBearer('Basic myToken123');
console.log(invalidToken); // null

alimd committed:
refactor(pre-handlers): Remove logger module

alimd committed:
refactor(pre-handlers): Update main.ts exports and add package tracer
